src/Impl/Utils/MultipleJsonsSplitter.php

```diff
@@ -0,0 +1,56 @@
+<?php declare(strict_types=1);
+
+namespace Streamx\Clients\Ingestion\Impl\Utils;
+
+class MultipleJsonsSplitter {
+
+    private const ESCAPED_QUOTE = '\"';
+    private const ESCAPED_QUOTE_TEMPORARY_REPLACEMENT = '^kj-N2hpS.';
+
+    /**
+     * @param $jsonsString string containing one or more JSON strings. They can be separated with any or no separator
+     * @return array of
+     */
+    static function splitToSingleJsons(string $jsonsString): array {
+        $resultJsons = [];
+        $openedBracesCounter = 0;
+        $currentJsonStartIndex = 0;
+        $insideQuotes = false;
+
+        $escapedJsonsString = self::maskEscapedQuotes($jsonsString);
```

Comment:
added this. Remove all instances of `\"` in the string, to not have to care about them for now